### ISSUE TITLE
Hitting GitHub rate limit when using vscode.dev for large repo

### ISSUE THREAD

**Issue by saragibby** [microsoft/vscode#164168]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 
- OS Version: 

Steps to Reproduce:

**Issue:** Looking to use vscode.dev in our student focused Web-Dev-for_Beginners repo (https://github.com/microsoft/Web-Dev-For-Beginners). With all lessons and translations this repo is about 1000 files. With minimal navigation in repo (acting as if I am a new student user), a user can hit the GitHub rate limit of 5000 and me unable to move forward in vscode.dev, or any other app utilizing the GitHub GraphQL API with the user's token.

_Note: with below steps I used GitHub CLI to monitor rate limit increases. Also, as time lapsed with back and forth and taking notes, it seemed network calls were continuing to GitHub API with no user action._

1. Using GitHub CLI, authenticate and can see that used 1 call, for rate limit used of 0.
    ```json
      "rate": {
        "limit": 5000,
          "used": 1,
            "remaining": 4999,
      }
    ``` 
2. Open https://github.com/microsoft/Web-Dev-For-Beginners in vscode.dev. To reflect a new user experience be sure all cache is cleared if previously opened.  Oen no files in repo.
    ```json
      "rate": {
        "limit": 5000,
          "used": 277,
            "remaining": 4723,
      }
    ``` 
3. Open the readme in main repo folder.
     ```json
      "rate": {
        "limit": 5000,
          "used": 834,
            "remaining": 4166,
      }
    ``` 
4. Preview readme file in vscode.dev
     ```json
      "rate": {
        "limit": 5000,
          "used": 1419,
            "remaining": 3581,
      }
    ``` 
5. Expand first lesson (getting startes -> intro to programming languages).
    ```json
      "rate": {
        "limit": 5000,
          "used": 2365,
            "remaining": 2636,
      }
    ``` 
6. View and then rpeview, both lesson 1 readme and assignment markdown files. 
    ```json
      "rate": {
        "limit": 5000,
          "used": 4260,
            "remaining": 740,
      }
    ``` 
7. Expand lesson 1 translations
    ```json
      "rate": {
        "limit": 5000,
          "used": 5000,
            "remaining": 0,
      }
    ``` 

WIth rate limit now met, I am now unable to view any files, or any other action with GitHub (i.e., create a branch or commit).

First error a user gets is:
![image](https://user-images.githubusercontent.com/82035/197034511-8a25e543-6175-4de0-8a78-25967487b9cb.png)

Upon reauthenticating:
![image](https://user-images.githubusercontent.com/82035/197034570-1c4fd470-d299-42f0-b4c3-951616fbe966.png)

Attempting to open a different file not previously opened:
![image](https://user-images.githubusercontent.com/82035/197034691-e85f2bc0-4e12-4d08-84f2-bbbe1b44bb8f.png)

**Comment by chrisdias** [comment]
how many files do you have to open in vscode.dev before the rate limit is hit?  i've opened just about every file i can find and i dont run into this problem.  i wonder if it has anything to do with the github account (e.g., me === chrisdias). Do some accounts have lower rate limits than others?

**Comment by joyceerhl** [comment]
1. Are you using https://vscode.dev/github/microsoft/Web-Dev-For-Beginners or https://insiders.vscode.dev/github/microsoft/Web-Dev-For-Beginners? If it's vscode.dev, could you please try on insiders.vscode.dev and see if the same behavior repros? 
2. Are you checking on the rate limit for the REST APIs or GraphQL APIs? I believe there are separate rate limits for each
3. If you go into your extensions pane and disable all extensions except the GitHub Repositories and Remote Repositories extensions, does the problem persist?
4. Is there anything in the output channel for GitHub Repositories, GitHub Codespaces or GitHub Pull Requests?

**Comment by joyceerhl** [comment]
I followed your repro steps in insiders.vscode.dev and I see


1. Using GitHub CLI, authenticate and can see that used 1 call, for rate limit used of 0.
    ```json
    "resources": {
        "graphql": {
        "limit": 5000,
        "used": 1,
        "remaining": 4999,
        "reset": 1666377052
        },
    },
    "rate": {
      "limit": 5000,
      "used": 0,
      "remaining": 5000,
      "reset": 1666377079
    }
    ``` 
2. Open https://github.com/microsoft/Web-Dev-For-Beginners in vscode.dev. To reflect a new user experience be sure all cache is cleared if previously opened.  Oen no files in repo.
    ```json
    "resources": {
        "graphql": {
        "limit": 5000,
        "used": 9,
        "remaining": 4991,
        "reset": 1666377052
        },
    },
    "rate": {
        "limit": 5000,
        "used": 7,
        "remaining": 4993,
        "reset": 1666377178
    }
    ``` 
3. Open the readme in main repo folder.
     ```json
    "resources": {
        "graphql": {
        "limit": 5000,
        "used": 10,
        "remaining": 4990,
        "reset": 1666377052
        }
    },
    "rate": {
        "limit": 5000,
        "used": 8,
        "remaining": 4992,
        "reset": 1666377178
    }
    ``` 
4. Preview readme file in vscode.dev
     ```json
    "resources": {
        "graphql": {
        "limit": 5000,
        "used": 12,
        "remaining": 4988,
        "reset": 1666377052
        }
    },
    "rate": {
        "limit": 5000,
        "used": 9,
        "remaining": 4991,
        "reset": 1666377178
    }
    ``` 
5. Expand first lesson (getting startes -> intro to programming languages). (NB: I expanded all subdirectories of this folder including translations at this point)
    ```json
    "resources": {
      "graphql": {
        "limit": 5000,
        "used": 13,
        "remaining": 4987,
        "reset": 1666377052
      }
    },
    "rate": {
      "limit": 5000,
      "used": 9,
      "remaining": 4991,
      "reset": 1666377178
    }
    ``` 
6. View and then rpeview, both lesson 1 readme and assignment markdown files. 
    ```json
    "resources": {
      "graphql": {
        "limit": 5000,
        "used": 14,
        "remaining": 4986,
        "reset": 1666377052
      }
    },
    "rate": {
      "limit": 5000,
      "used": 11,
      "remaining": 4989,
      "reset": 1666377178
    }
    ```

**Comment by joyceerhl** [comment]
@chrisdias I think we do have elevated API rate limits, but even when testing with a throwaway GitHub account that doesn't have the elevated API rate limit, I do not manage to exhaust it. @saragibby are you signed into any other applications with GitHub, or do you have other extensions installed in vscode.dev which might be abusing the API? @alexr00 also just fixed a rate limit issue in GHPRI, but I doubt this is the same problem.

**Comment by alexr00** [comment]
The GHPRI rate limit issue occurred when you have the overview page of PR open and visible in VS Code (this includes vscode.dev).

**Comment by joyceerhl** [comment]
Just to close the loop here, @saragibby and I chatted about this over Teams last week and we cannot reproduce the api rate limit on either stable or insiders vscode.dev. In future if this happens again, the GitHub Repositories output channel log inside VS Code as well as the network trace (Toggle Developer Tools > Network tab) would help us figure out whether it's VS Code or an extension within it which is consuming the api quota. For now I'll close this out but please reopen if it happens again.

**Comment by saragibby** [comment]
@joyceerhl thank you for connecting with me on this! After we walked through the scenarios together, I tried to reproduce the rate limit issue in all varying scenarios with repos, browsers, and extensions. Including an external student's GitHub account (no connection to GitHub or Microsoft) and was unable to hit the rate limit in any scenario.

Even with 3 repos opened in VS Code locally, and 2 instances of vscode.dev with the Web Dev for Beginners curricula in different browsers I didn't hit the rate limit. Unclear what changed, but super excited to know we can integrate vscode.dev into our tutorials and curriculum without fear of getting the student into a rate limit issue. 

Thank you so much for your time and help on this!!
